extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -33,25 +35,36 @@ class AmpYoutube extends AMP.BaseElement {
   }
 
   /** @override */
-  layoutCallback() {
+  buildCallback() {
     const width = this.element.getAttribute('width');
     const height = this.element.getAttribute('height');
+
+    this.width_ = getLengthNumeral(width);
+    this.height_ = getLengthNumeral(height);
+
     // The video-id is supported only for backward compatibility.
-    const videoid = AMP.assert(
+    /** @private @const {!string} */
+    this.videoid_ = AMP.assert(
         (this.element.getAttribute('data-videoid') ||
         this.element.getAttribute('video-id')),
         'The data-videoid attribute is required for <amp-youtube> %s',
         this.element);
+
+    this.buildImagePlaceholder_();
```

Comment:
Only call when the tag does NOT have placeholder child: `!this.getPlaceholder()`